purchase_request_expense/wizard/purchase_request_line_make_expense.py

```diff
@@ -0,0 +1,320 @@
+            "name": line.name or product.name,
+            "product_qty": line.pending_qty_to_receive or 1.0,
+            "product_uom_id": line.product_uom_id.id or product.uom_id.id,
+            "estimated_cost": line.estimated_cost,
+            "analytic_account_id": line.analytic_account_id.id
+            if line.analytic_account_id
+            else False,
+        }
+
+    @api.model
+    def _check_valid_request_line(self, request_line_ids):
+        picking_type = False
+        company_id = False
+
+        for line in self.env["purchase.request.line"].browse(request_line_ids):
+            if line.request_id.state == "done":
+                raise UserError(_("The purchase has already been completed."))
+            if line.request_id.state != "approved":
+                raise UserError(
+                    _("Purchase Request %s is not approved") % line.request_id.name
+                )
+
+            if line.purchase_state == "done":
+                raise UserError(_("The purchase has already been completed."))
+
+            line_company_id = line.company_id and line.company_id.id or False
+            if company_id is not False and line_company_id != company_id:
+                raise UserError(_("You have to select lines from the same company."))
+            else:
+                company_id = line_company_id
+
+            line_picking_type = line.request_id.picking_type_id or False
+            if not line_picking_type:
+                raise UserError(_("You have to enter a Picking Type."))
+            if picking_type is not False and line_picking_type != picking_type:
+                raise UserError(
+                    _("You have to select lines from the same Picking Type.")
+                )
+            else:
+                picking_type = line_picking_type
+
+    @api.model
+    def check_group(self, request_lines):
+        if len(list(set(request_lines.mapped("request_id.group_id")))) > 1:
+            raise UserError(
+                _(
+                    "You cannot create a single purchase order from "
+                    "purchase requests that have different procurement group."
+                )
+            )
+
+    @api.model
+    def get_items(self, request_line_ids):
+        request_line_obj = self.env["purchase.request.line"]
+        items = []
+        request_lines = request_line_obj.browse(request_line_ids)
+        self._check_valid_request_line(request_line_ids)
+        self.check_group(request_lines)
+        for line in request_lines:
+            item = self._prepare_item(line)
+            if item:
+                items.append([0, 0, item])
+        if not items:
+            raise UserError(
+                _("No products that can be expensed were found in the selected lines.")
+            )
+        return items
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        active_model = self._context.get("active_model", False)
```

Comment:
```suggestion
        active_model = self.env.context.get("active_model", False)
```

`_context` is deprecated for a long time